platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -167,33 +175,57 @@ public void destroy() throws ApplicationServiceException {
     }
   }
 
+  @Override
+  public void installProfile(String profile, boolean noRefresh) {
+    try {
+      EnumSet<FeaturesService.Option> options = EnumSet.noneOf(FeaturesService.Option.class);
+      if (noRefresh) {
+        options.add(FeaturesService.Option.NoAutoRefreshBundles);
+      }
+      featuresService.installFeature(profile, options);
+    } catch (Exception e) {
+      LOGGER.error("Could not start profile feature: {}", profile, e);
+      throw new ApplicationServiceBeanException("Could not start profile feature: " + profile);
+    }
+  }
+
   @Override
   public List<Map<String, Object>> getInstallationProfiles() {
-    List<Feature> installationProfiles = appService.getInstallationProfiles();
-    List<Map<String, Object>> profiles = new ArrayList<>();
+    try {
+      List<Feature> installationProfiles = appService.getInstallationProfiles();
+      List<Map<String, Object>> profiles = new ArrayList<>();
 
-    for (Feature profile : installationProfiles) {
-      Map<String, Object> profileMap = new HashMap<>();
-      profileMap.put(INSTALL_PROFILE_NAME, profile.getName());
-      profileMap.put(INSTALL_PROFILE_DESCRIPTION, profile.getDescription());
+      for (Feature profile : installationProfiles) {
+        Map<String, Object> profileMap = new HashMap<>();
+        profileMap.put(INSTALL_PROFILE_NAME, profile.getName());
+        profileMap.put(INSTALL_PROFILE_DESCRIPTION, profile.getDescription());
 
-      List<String> includedFeatures = new ArrayList<>();
-      profile.getDependencies().forEach(dep -> includedFeatures.add(dep.getName()));
-      profileMap.put(INSTALL_PROFILE_DEFAULT_APPLICATIONS, includedFeatures);
+        List<String> includedFeatures = new ArrayList<>();
+        profile.getDependencies().forEach(dep -> includedFeatures.add(dep.getName()));
+        profileMap.put(INSTALL_PROFILE_DEFAULT_APPLICATIONS, includedFeatures);
 
-      profiles.add(profileMap);
-    }
+        profiles.add(profileMap);
+      }
+      return profiles;
 
-    return profiles;
+    } catch (Exception e) {
+      LOGGER.error("Could not get installation profiles", e);
+      throw new ApplicationServiceBeanException("Could not get installation profiles");
+    }
   }
 
   @Override
   public List<Map<String, Object>> getApplications() {
-    return appService
-        .getApplications()
-        .stream()
-        .map(this::convertApplicationEntries)
-        .collect(Collectors.toList());
+    try {
+      return appService
+          .getApplications()
+          .stream()
+          .map(this::convertApplicationEntries)
+          .collect(Collectors.toList());
+    } catch (Exception e) {
```

Comment:
These catch code blocks seem to be very repetitive, should they be factored out into a utility function/lambda?